InMemoryRepositories/ShiftInMemoryRepository.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Entities;
+using RepositoryContracts;
+
+namespace InMemoryRepositories;
+
+public class ShiftInMemoryRepository : IShiftRepository
+{
+    private List<Shift> shifts = new List<Shift>();
+    
+    public async Task<Shift> AddAsync(Shift shift)
+    {
+        shifts.Add(shift);
+        return shift;
+    }
+
+    
+    public async Task<Shift> UpdateAsync(Shift shift)
+    {
+        Shift? exisitingShift = shifts.SingleOrDefault(p => p.Id == shift.Id);
```

Comment:
what is SingleOrDefault and p in it?

Reply:
it is a getter method based on p(new general object [shift rn cuz we are in shifts list]) and we r looking for id that matches from parameter